drafts/angular-navigation-concept.md

```diff
@@ -0,0 +1,607 @@
+      this.historyGo(relativePosition * -1);
+    } else {
+      this.router.navigate(['/home']);
+    }
+  }
+
+  // ...
+}
+```
+
+## 1. Getting the Offsets
+
+The offsets are stored within history state, so we only need to read it to get the current values.
+Depending on the `target`, we then access the corresponding offset.
+
+## 2. Wrong Usage
+
+If the offset is 0, it typically indicates a programming error.
+To make the programmer aware of the issue, an error is logged.
+It is a programming mistake, because e.g. `navigateBackTo('flow-source-page')` should not be called on a `flow source page`.
+
+## 3. Navigation
+
+When an offset is found, the user is navigated back by the specified number of pages.
+
+However, logged-in users can directly access the application on a detail page, which may not be a `flow source page` or a `main page`.
+In such cases, the offsets would not be set.
+Instead of preventing navigation, the fallback navigates the user to the home page.
+
+# Requirement 2: Navigate Back Over Finished Flows
+
+Before diving into the solution, it is necessary to cover some `Angular Router` basics.
+Let us briefly look at the configuration used in this project.
+
+## Angular Router Theory
+
+```ts
+withRouterConfig({ canceledNavigationResolution: 'computed', onSameUrlNavigation: 'ignore' });
+```
+
+### canceledNavigationResolution
+
+The `canceledNavigationResolution` setting defines what should happen after a navigation is canceled.
+
+- With `replace (default)`, the canceled navigation URL will be replaced by the origin.
+  ![Replace cancel navigation resolution](https://cdn.hashnode.com/res/hashnode/image/upload/v1731316497961/8B2GoQVRe.png?auto=format)
+- With `computed`, the `Angular Router` attempts to restore the state before the canceled navigation.
+  This keeps the browser history stack intact.
+  ![Compute cancel navigation resolution](https://cdn.hashnode.com/res/hashnode/image/upload/v1731316514823/iO7cKQORz.png?auto=format)
+
+### onSameUrlNavigation
+
+The `onSameUrlNavigation` setting defines what should happen when the same URL is activated, for example, if the user clicks the same menu item twice.
+
+- `ignore (default)`: The navigation will be skipped.
+- `reload`: The router will reload the component and re-execute all guards.
+
+## Theory
+
+The first problem to address is detecting a back navigation that would lead into a finished flow.
+![Example of detecting flows](https://cdn.hashnode.com/res/hashnode/image/upload/v1735736268084/Yh6GI65zT.png?auto=format)
+
+Once this scenario is detected, navigation can proceed to the last `flow source page`, which we implemented in the previous step.
+
+## Detecting Navigation into Finished Flow
+
+Detecting back navigation into a finished flow can be solved using a guard: `flow-page-activation.guard`.
+The guard verifies if the origin URL belongs to the current flow.
+If it does, the navigation is allowed; otherwise, the navigation is blocked.
```

Comment:
That means that the "flow confirmation page" aka the page where the flow is finished, must not have the `flowBasePath`. Do I understand this correctly?